packages/esm-patient-registration-app/src/patient-registration/field/address/address-field.component.tsx

```diff
@@ -23,42 +24,49 @@ export const AddressField: React.FC = () => {
   const { addressTemplate } = useContext(ResourcesContext);
   const [addressFields, setAddressFields] = useState([]);
   const { t } = useTranslation();
-  const addressTemplateXml = addressTemplate.results[0].value;
+  const addressTemplateXml = addressTemplate?.results[0].value;
 
   useEffect(() => {
-    const templateXmlDoc = parseString(addressTemplateXml);
-    const nameMappings = getTagAsDocument('nameMappings', templateXmlDoc);
-    const elementDefaults = getTagAsDocument('elementdefaults', templateXmlDoc);
-    const properties = nameMappings.getElementsByTagName('property');
-    const propertiesObj = Array.prototype.map.call(properties, (property: Element) => {
-      const name = property.getAttribute('name');
+    if (addressTemplateXml) {
+      const templateXmlDoc = parseString(addressTemplateXml);
+      const nameMappings = getTagAsDocument('nameMappings', templateXmlDoc);
+      const elementDefaults = getTagAsDocument('elementdefaults', templateXmlDoc);
+      const properties = nameMappings.getElementsByTagName('property');
+      const propertiesObj = Array.prototype.map.call(properties, (property: Element) => {
+        const name = property.getAttribute('name');
+        const labelText = t(name, property.getAttribute('value'));
+        const value = getFieldValue(name, elementDefaults);
+        return {
+          id: name,
+          name,
+          labelText,
+          value,
+        };
+      });
+      setAddressFields(propertiesObj);
+      // Do not remove the comments below
       /*
         t('postalCode')
         t('address1')
         t('stateProvince')
         t('cityVillage')
         t('country')
       */
```

Comment:
We should move these comments with the `labelText` line, since these comments document the translation keys we expect to support.

Reply:
Hi @ibacher!
Will the place matter? The reason is because whenever the translation key is called by the `t` hook, it looks into the `en.json` file, hence if the key is present in the `en.json`, there won't be any reason left for these keys to be placed near the labelText, what do you say?